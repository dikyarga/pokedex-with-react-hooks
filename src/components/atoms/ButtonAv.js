import React from "react";

function ButtonAtomView({
  onClicked,
  children,
  isPrimary = false,
  className = ""
}) {
  const buttonClass = isPrimary
    ? "bg-blue-500 hover:bg-blue-400 text-white font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded m-2"
    : "bg-white hover:bg-gray-100 text-gray-800 font-semibold py-2 px-4 border border-gray-400 rounded shadow m-2";
  return (
    <button className={`${className} ${buttonClass}`} onClick={onClicked}>
      {children}
    </button>
  );
}

export default ButtonAtomView;
