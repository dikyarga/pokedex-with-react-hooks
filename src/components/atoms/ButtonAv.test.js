import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";

import ButtonAv from "./ButtonAv";

afterEach(cleanup);

describe("<ButtonAv />", () => {
  const content = "content";
  const onClicked = jest.fn();

  it("Should fire onClicked prop", () => {
    const { getByText } = render(
      <ButtonAv onClicked={onClicked}>{content}</ButtonAv>
    );
    const counter = getByText(content);

    fireEvent.click(counter);

    expect(onClicked).toHaveBeenCalled();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(<ButtonAv>Text</ButtonAv>);
    expect(asFragment()).toMatchSnapshot();
  });
});
