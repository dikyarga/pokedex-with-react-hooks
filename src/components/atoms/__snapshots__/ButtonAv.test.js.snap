// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonAv /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class=" bg-white hover:bg-gray-100 text-gray-800 font-semibold py-2 px-4 border border-gray-400 rounded shadow m-2"
  >
    Text
  </button>
</DocumentFragment>
`;
